storage/innobase/log/log0online.cc

```diff
@@ -1897,14 +2084,14 @@ bool log_online_purge_changed_page_bitmaps(
   }
 
   if (log_bmp_sys_inited) {
-    if (lsn > log_bmp_sys->end_lsn) {
+    if (log_bmp_sys->checkpoint_lsn != 0 && lsn > log_bmp_sys->checkpoint_lsn) {
       lsn_t new_file_lsn;
       if (lsn == LSN_MAX) {
         /* RESET restarts the sequence */
         log_bmp_sys->out_seq_num = 0;
         new_file_lsn = 0;
       } else {
-        new_file_lsn = log_bmp_sys->end_lsn;
+        new_file_lsn = log_bmp_sys->checkpoint_lsn;
```

Comment:
shouldn't the end_lsn be the parse_stop_lsn?

Reply:
Good point, will recheck this.

Reply:
We can use log_bmp_sys->start_lsn to get last parsed LSN from previous parser iteration. We store parse_buf.parse_stop_lsn there to keep state between iterations. I've updated this part a bit.